lib/arc_ecto/schema.ex

```diff
@@ -53,4 +22,37 @@ defmodule Arc.Ecto.Schema do
 
     end) || params
   end
+
+  # If given a changeset, apply the changes to obtain the underlying data
+  # It would be nice to know under what context this will not be a changeset.
+  defp normalize_scope changeset_or_data do
+    case changeset_or_data do
+      %Ecto.Changeset{} -> Ecto.Changeset.apply_changes(changeset_or_data)
+      %{__meta__: _} -> changeset_or_data #TODO: wont this cause normalize_params to fail?
```

Comment:
Sorry I should have been more clear. I mean under what context would `cast_attachemnts` get a struct vs a changeset. From what I can tell it will always be a changeset, but that being said I'm completely new to all this.